### ISSUE TITLE
Class cast exception for including Medication in MedicationRequest via custom-queries

### ISSUE THREAD

**Issue by kmarx** [drajer-health/eCRNow#584]
To reduce the volume of FHIR requests for `Medication` we have a custom queries entry of

`MedicationRequest?patient=Patient/{{context.patientId}}&_include=MedicationRequest:medication`

However this seems to have stopped working at least as of the 2023-02-03 ERSD file. (Not sure that's really related.) We emit a bundle with both resources in it for the `MedicationRequest` query, and eCRNow gets a class cast exception trying to convert one of the "_include"d `Medications` to a `MedicationRequest`.

Fwiw, we're on R4 FHIR.

```
{
    "resourceType": "Bundle",
    "id": "92eac4fe-e8ff-45fd-9c33-bec030890833",
    "meta": {
        "lastUpdated": "2023-09-14T05:44:22.607-07:00"
    },
    "type": "searchset",
    "total": 2,
    "link": [
        {
            "relation": "self",
            "url": "http://localhost:8090/xfhir/MedicationRequest?_include=MedicationRequest%3Amedication&patient=Patient%2F892CB66DA43E419FAD21D91182FB3A4E"
        }
    ],
    "entry": [
        {
            "resource": {
                "resourceType": "MedicationRequest",
                "id": "5630731",
				.....
            }
        },
        {
            "resource": {
                "resourceType": "MedicationRequest",
                "id": "5630733",
                .......
            }
        },
        {
            "resource": {
                "resourceType": "Medication",
                "id": "12479-1",
                ........
            }
        }
    ]
}
```

```
14-09-2023 11:42:33.526 [35m[db-scheduler-pool-1-thread-4][0;39m [1;31mERROR[0;39m c.d.b.service.impl.KarProcessorImpl.applyKarForScheduledJob - Exception encountered during processing of the scheduled job 
14-09-2023 11:42:33.535 [35m[db-scheduler-pool-1-thread-4][0;39m [1;31mERROR[0;39m c.d.ecrapp.util.ApplicationUtils.handleException - Error in completing the Execution, retry in 5 minutes.
java.lang.ClassCastException: class org.hl7.fhir.r4.model.Medication cannot be cast to class org.hl7.fhir.r4.model.MedicationRequest (org.hl7.fhir.r4.model.Medication and org.hl7.fhir.r4.model.MedicationRequest are in unnamed module of loader 'app')
	at com.drajer.bsa.utils.R3ToR2DataConverterUtils.addResourcesToR4FhirData(R3ToR2DataConverterUtils.java:274)
	at com.drajer.bsa.utils.R3ToR2DataConverterUtils.convertKarProcessingDataForCdaGeneration(R3ToR2DataConverterUtils.java:105)
	at com.drajer.bsa.kar.action.EcrReportCreator.getCdaR11Report(EcrReportCreator.java:353)
	at com.drajer.bsa.kar.action.EcrReportCreator.createReport(EcrReportCreator.java:161)
	at com.drajer.bsa.kar.action.EcrReportCreator.createReport(EcrReportCreator.java:122)
	at com.drajer.bsa.kar.action.CreateReport.process(CreateReport.java:106)
	at com.drajer.bsa.kar.model.BsaAction.executeRelatedActions(BsaAction.java:245)
	at com.drajer.bsa.kar.action.CheckTriggerCodes.process(CheckTriggerCodes.java:129)
	at com.drajer.bsa.kar.model.BsaAction.executeSubActions(BsaAction.java:162)
	at com.drajer.bsa.kar.action.ExecuteReportingActions.process(ExecuteReportingActions.java:32)
	at com.drajer.bsa.service.impl.KarProcessorImpl.applyKarForScheduledJob(KarProcessorImpl.java:247)
	at com.drajer.bsa.service.impl.KarProcessorImpl$$FastClassBySpringCGLIB$$71cc7050.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:367)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:118)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)
	at com.drajer.bsa.service.impl.KarProcessorImpl$$EnhancerBySpringCGLIB$$5149b260.applyKarForScheduledJob(<generated>)
	at com.drajer.bsa.scheduler.ScheduleJobConfiguration.lambda$sampleOneTimeJob$0(ScheduleJobConfiguration.java:58)
	at com.github.kagkarlsson.scheduler.task.helper.Tasks$OneTimeTaskBuilder$1.executeOnce(Tasks.java:211)
	at com.github.kagkarlsson.scheduler.task.helper.OneTimeTask.execute(OneTimeTask.java:52)
	at com.github.kagkarlsson.scheduler.ExecutePicked.executePickedExecution(ExecutePicked.java:85)
	at com.github.kagkarlsson.scheduler.ExecutePicked.run(ExecutePicked.java:68)
	at com.github.kagkarlsson.scheduler.FetchCandidates.lambda$run$1(FetchCandidates.java:89)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at com.github.kagkarlsson.scheduler.FetchCandidates.lambda$run$2(FetchCandidates.java:87)
	at com.github.kagkarlsson.scheduler.Executor.lambda$addToQueue$0(Executor.java:52)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
14-09-2023 11:42:33.536 [35m[db-scheduler-pool-1-thread-4][0;39m [34mINFO [0;39m c.d.b.s.ScheduleJobConfiguration.lambda$sampleOneTimeJob$0 -  Clearing MDC
14-09-2023 11:42:33.536 [35m[db-scheduler-pool-1-thread-4][0;39m [31mWARN [0;39m c.g.kagkarlsson.scheduler.Scheduler.log - Unhandled exception during execution of task with name 'BsaScheduledJob'. Treating as failure.
```

**Comment by ugaturk95** [comment]
Hello kmarx, 
**Could you please send a recent FHIR Loading Query demonstrating the issue?**  This will help us debug end to end.  You can send it in an encrypted email to me @ shaturk@deloitte.com.  We need one for the last few days because the eICR XML content only remains in AWS S3 for 7 days.
Thanks,
Shawn Turk

**Comment by Balakrishnan-369** [comment]
Hi Shawn,
It appears that the LoadingQueryBundle is not generated when this particular issue occurs. In light of this, I have emailed you the generated TriggerQueryBundle file for analysis.

**Comment by nbashyam** [comment]
Fixed in 3.1.3
